microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/ls/ProjectTypesDelegateCommandHandler.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+* Copyright (c) 2019 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.core.ls;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.ls.core.internal.IDelegateCommandHandler;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
+
+import com.redhat.microprofile.commons.ProjectTypeResults;
+import com.redhat.microprofile.jdt.core.ProjectTypeDefinition;
+import com.redhat.microprofile.jdt.internal.core.ProjectTypeRegistry;
+
+
+/**
+ * Delegate command handler to determine project types in the workspace
+ * 
+ */
+public class ProjectTypesDelegateCommandHandler implements IDelegateCommandHandler {
+
+	private static final String PROJECT_TYPE_COMMAND_ID = "java/projectTypes";
+	
+	private static final String MAVEN_TYPE = "maven";
+
+	private static final String GRADLE_TYPE = "gradle";
+	
+	private static final String UNKNOWN_TYPE = "unknown";
+
+
+	public ProjectTypesDelegateCommandHandler() {
+	}
+
+	@Override
+	public Object executeCommand(String commandId, List<Object> arguments, IProgressMonitor progress) throws Exception {
+		switch (commandId) {
+		case PROJECT_TYPE_COMMAND_ID:
+			return getProjectTypeResults();
+		default:
+			throw new UnsupportedOperationException(String.format("Unsupported command '%s'!", commandId));
+		}
+	}
+	
+	private static ProjectTypeResults getProjectTypeResults() {
+		ProjectTypeResults results = new ProjectTypeResults();
+		IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+		
+		for (IProject project: projects) {
+			try {
+				IProjectDescription description = project.getDescription(); 
+				URI uri = description.getLocationURI();
+				String path = uri.getPath();
+				if (uri != null) {
+					results.addResult(path, getProjectType(project));
+				}
+			} catch (CoreException|NullPointerException e) {
+				// ignore
+			}
+		}
+		return results;
+	}
+	
+	private static String getProjectType(IProject project) {
```

Comment:
If I understand the code, this method could return quarkus, MAVEN, GRADLE, UNKWOWN_TYPE. The project type is strange because quarkus project type can be a maven project or a gradle project too.

Perhaps we should return the both information quarkus (as kind project) and MAVEN (as type project) for instance. @fbricon what do you think about that?